### PR TITLE
[DeviceSanitizer] Handle mem buffer is created with flag COPY_HOST_POINTER

### PR DESCRIPTION
**Pull request by zhaomaosu**
SYCLOS Part for pre-ci: https://github.com/intel/llvm/pull/14551

### CODE REVIEW

**Review by AllanZyne — APPROVED** [review]

**Comment by zhaomaosu** [comment]
Hi @oneapi-src/unified-runtime-maintain, could you please help review this PR? We need this bug fix PR in v0.10.0rc release branch. Thanks.

**Comment by zhaomaosu** [comment]
> Hi @oneapi-src/unified-runtime-maintain, could you please help review this PR? We need this bug fix PR in v0.10.0rc release branch. Thanks.

Kindly ping @oneapi-src/unified-runtime-maintain

File: source/loader/layers/sanitizer/ur_sanddi.cpp [hunk 1]
```diff
@@ -432,6 +432,19 @@ __urdlllocal ur_result_t UR_APICALL urMemBufferCreate(
 
     std::shared_ptr<MemBuffer> pMemBuffer =
         std::make_shared<MemBuffer>(hContext, size, hostPtrOrNull);
+
+    if (Host && (flags & UR_MEM_FLAG_ALLOC_COPY_HOST_POINTER)) {
+        std::shared_ptr<ContextInfo> CtxInfo =
+            getContext()->interceptor->getContextInfo(hContext);
+        for (const auto &hDevice : CtxInfo->DeviceList) {
+            ManagedQueue InternalQueue(hContext, hDevice);
+            char *Handle;
```
Comment (aarongreig) [inline source/loader/layers/sanitizer/ur_sanddi.cpp:441]: I suspect coverity might complain about this so might as well initialize it
```suggestion
            char *Handle = nullptr;
```

**Review by aarongreig — CHANGES_REQUESTED** [review]
LGTM pending comment

File: source/loader/layers/sanitizer/ur_sanddi.cpp (hunk 1 above)
Comment (zhaomaosu) [reply source/loader/layers/sanitizer/ur_sanddi.cpp]: Thanks. Done.

**Review by aarongreig — APPROVED** [review]

**Pull request merged**